lib/src/lifecycle_module.dart

```diff
@@ -130,12 +156,28 @@ abstract class LifecycleModule extends Object with Disposable {
       _didUnloadChildModuleSubscriptions = {};
 
   /// Whether the module is currently loaded.
-  bool get isLoaded => _isLoaded;
+  bool get isLoaded => _state == LifecycleState.loaded;
+
+  /// Whether the module is currently loading.
+  bool get isLoading => _state == LifecycleState.loading;
+
+  /// Whether the module is currently resuming.
+  bool get isResuming => _state == LifecycleState.resuming;
 
   /// Whether the module is currently suspended.
-  ///
-  /// This will always be false when the module is not loaded.
-  bool get isSuspended => _isSuspended;
+  bool get isSuspended => _state == LifecycleState.suspended;
+
+  /// Whether the module is currently suspending.
+  bool get isSuspending => _state == LifecycleState.suspending;
+
+  /// Whether the module is currently unloaded.
+  bool get isUnloaded => _state == LifecycleState.unloaded;
+
+  /// Whether the module is currently unloading.
+  bool get isUnloading => _state == LifecycleState.unloading;
+
+  /// The current lifecycle state of the module.
+  LifecycleState get state => _state;
```

Comment:
Personally I lean towards not including this `state` property because we don't have a use case for it (outside of tests) yet and it forces us to commit to `LifecycleState` as a part of the public API. If we don't do that, then we're only committing to the `bool is[State]` getters.

It's just a lean, though. If others think this adds enough value, I'm cool with it.